### PR TITLE
Converting to select2 v 4

### PR DESCRIPTION
**Pull request by edster**
Notes:

With the way select2 version 4 does there scaffolding I am concerned with hiding values from the search if it is selected.

If we hide them it will be hidden regardless if multiple is used or not. This can or cannot be a concern.

I was also thinking we could use the bootstrap theme and theme it from there if you don't mind changing the way the tagging works a bit.

I made the css apply as the default ALL select2's on a page rather then a theme, this will allow others who implement it to apply there own theme to change certain aspects without having to copy over all the styling.

Can apply it to core, or keep it as multiple only as it seems multiple is applied to every page anyways.

### CODE REVIEW

**Comment by edster** [comment]
Wasn't sure best way to do it but was thinking we should attach a function to call the function attached to .select2-selection__choice__remove when .select2-selection__choice is pressed.

(Will fix the issue that only pressing the x removes the choice, but that x is super small)

File: resources/views/tags.twig [hunk 1]
```diff
@@ -1,8 +1,9 @@
+{{ asset_add("styles.css", "anomaly.field_type.multiple::css/select2.min.css") }}
```
Comment (RyanThompson) [inline resources/views/tags.twig:1]: We need both of these includes?

Comment (edster) [reply resources/views/tags.twig:1]: Line 1 is the select2 css that comes from the vendor, 
Line 2 is the styling trying to match the theming to your old one.

**Comment by RyanThompson** [comment]
Still some ugliness going on. They should look JUST like they do now in the non-v4 select2

screen shot 2016-09-29 at 4 05 31 pm
screen shot 2016-09-29 at 4 05 36 pm
screen shot 2016-09-29 at 4 05 44 pm

**Comment by RyanThompson** [comment]
For example the borders / shadows and the tag makes the input grow past it's original size.

**Comment by edster** [comment]
I’ll do some more digging. There are limitations on some things with the new version in regards to styling.

Regards,

Edi Mange

**Comment by huglester** [comment]
@edster any luck maybe?

**Comment by RyanThompson** [comment]
I am going to end up tackling this in the UI update.

**Pull request closed without merging**